### ISSUE TITLE
Lab VM & Sample deployment failed

### ISSUE THREAD

**Issue by fyunis** [microsoft/MCW-Building-a-resilient-IaaS-architecture#96]
Hi,
With the latest updates, I have worked on the updated and started testing the lab.

In before HOL, the sample deployment with error in SQL Server/JoinDomain.
Details:

 "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "VMExtensionProvisioningError",
      "message": "VM has reported a failure when processing extension 'JoinDomain'. Error message: \"Exception(s) occured while joining Domain 'contoso.com'\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/vmextensionwindowstroubleshoot "
    }
  ]
}

And the LabVM failed with the error:

  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "VMExtensionProvisioningError",
      "message": "VM has reported a failure when processing extension 'CustomScriptExtension1'. Error message: \"Finished executing command\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSEWindowsTroubleshoot "
    }

please your support!

**Comment by shantha05** [comment]
Hi Fyunis

Thanks for reporting the issue. Sorry for the delayed response. Currently I am working on a fix for this issue. I will keep you posted.

Thanks
Anu

**Comment by sadukie** [comment]
Confirming that in running BHOL, LabVM/CustomScriptExtension1, I am also seeing the failure with this message:

```
{
    "status": "Failed",
    "error": {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'CustomScriptExtension1'. Error message: \"Finished executing command\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/VMExtensionCSEWindowsTroubleshoot "
    }
}
```

This does create the following successfully:
- LabVM
- LabVMNIC
- LabVMVNey
- LabVMNSG
- LabVMIP

@timahenning Looking into this one

**Comment by kalyani7744** [comment]
Hello, I was working on the latest updates and I am facing the same issue while deploying ContosoRG1.

In before HOL, the deployment is failing with an error in SQL Server/JoinDomain.
Error Details:

{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "VMExtensionProvisioningError",
      "message": "VM has reported a failure when processing extension 'JoinDomain'. Error message: \"Exception(s) occured while joining Domain 'contoso.com'\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/vmextensionwindowstroubleshoot "
    }
  ]
}

![image](https://user-images.githubusercontent.com/83349665/168832010-d22d8544-4755-4f57-b32b-61b9934cd006.png)


Could you please look into it and provide an update?

**Comment by DawnmarieDesJardins** [comment]
Reopening issue for further investigation.

**Comment by DawnmarieDesJardins** [comment]
@kalyani7744 - Thanks for the information.  We are almost ready to merge an author review of the workshop with updates that may fix this issue.  Please hold, we'll have that out soon.

**Comment by benstegink** [comment]
@kalyani7744 - This does appear to be a bit random based on how "fast" certain steps are happening when deploying the resources. Until the fix is pushed out, you could try a couple of things as well:

1. Just delete the resources in your subscription and try again. Sometimes that works.
2. Try a region other than Central US. Personally, I've had better luck with East US 2 or West US 2.

**Comment by kalyani7744** [comment]
Hello @benstegink, I have tried the deployment in East US2 and West US2 Regions as you suggested and the deployments got succeeded.

While testing the lab, in [Exercise 1, Task 3, step 48](https://github.com/microsoft/MCW-Building-a-resilient-IaaS-architecture/blob/master/Hands-on%20lab/HOL%20step-by%20step%20-%20Building%20a%20resilient%20IaaS%20architecture.md#task-3-configure-ha-for-the-sql-server-tier) we are facing an error while adding the availability group in SQLVM1. The availability group validation is getting failed with the following errors.

Please find the error details below:

![image](https://user-images.githubusercontent.com/83349665/169232855-4ebe35ca-f10b-4aa2-9bc3-c270d04a892e.png)

![image](https://user-images.githubusercontent.com/83349665/169232899-eb058edf-f93f-4b85-b8a3-d5078fbfad0a.png)

![image](https://user-images.githubusercontent.com/83349665/169232943-0b0a6238-16db-4e8f-94b9-46809e642800.png)


- While troubleshooting we found that in SQLVM2 the **Contoso Insurance** database is already created. We have deleted the database and revalidated the availability group creation. The validations got succeeded after that.

![image](https://user-images.githubusercontent.com/83349665/169235218-2fdf9449-b5f8-4699-8350-070b168b65d8.png)

We have an upcoming lab, could you please look into this and provide an update?

**Comment by benstegink** [comment]
@kalyani7744 yup, that was another fix that's coming in the upcoming change. You can just manually delete the contosoinsurance database from SQLVM2 (right click -> delete). Then you should be able to keep moving forward with the lab. When you get toe SQLVM3 you may see the same issue and may need to manually delete the database on that VM as well.

**Comment by DawnmarieDesJardins** [comment]
Closing issue with merge of test/fix PR #100
